### PR TITLE
feat: hexbytes validation w/ pydantic

### PR DESCRIPTION
**Pull request by NotPeopling2day**
### What I did
Whenever you wanted to use HexBytes as a type with a pydantic model you would get a validation error, now HexBytes can be validated and serialized on a pydantic model.

fixes: #20 

### How I did it
Used the HexBytes class as a BaseModel, adding pydantic validation models
Used the Pydantic BaseModel class but specialized the `_get_value()` method to return `hex` when serializing.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by NotPeopling2day** [comment]
Doing this as a proof of concept where we can use these across our core without having to rename everything, let me know your thoughts.

File: ethpm_types/utils.py [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 
 
+class HexBytes(BaseHexBytes):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return HexBytes(v)
```
Comment (fubuloubu) [inline ethpm_types/utils.py:16]: Would this work?

```suggestion
        yield cls
```

**Review by fubuloubu — APPROVED** [review]

File: ethpm_types/utils.py (hunk 1 above)
Comment (NotPeopling2day) [reply ethpm_types/utils.py:16]: It doesn't, it then would return the actual class. `{'byte_bytes': <class 'ethpm_types.utils.HexBytes'>, ...}`

Comment (fubuloubu) [reply ethpm_types/utils.py:16]: Weird, I guess it does have to be a class method like `lambda cls, v: ...`

File: ethpm_types/base.py [hunk 2]
```diff
@@ -33,3 +37,31 @@ def json(self, *args, **kwargs) -> str:
             kwargs["exclude_none"] = True
 
         return super().json(*args, **kwargs)
+
+    @classmethod
+    @no_type_check
+    def _get_value(
+        cls,
+        v: Any,
+        to_dict: bool,
```
Comment (unparalleled-js) [inline ethpm_types/base.py:46]: I wonder if you could just use args and kwargs for these? (Except `v` of course)

File: tests/test_hexbytes.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from ethpm_types import BaseModel, HexBytes
+
+TEST_MODEL_DICT = {
```
Comment (unparalleled-js) [inline tests/test_hexbytes.py:3]: Should we include a more filled out `HexBytes` value in here? Like an address?

**Review by unparalleled-js — APPROVED** [review]

File: tests/test_hexbytes.py (hunk 3 above)
Comment (NotPeopling2day) [reply tests/test_hexbytes.py:3]: If you think so? I was just trying to show that I didn't break the functionality of HexBytes or Pydantic by adding the validation and serialization methods.

File: ethpm_types/base.py (hunk 2 above)
Comment (NotPeopling2day) [reply ethpm_types/base.py]: Updated [70ab8e2](https://github.com/ApeWorX/ethpm-types/pull/33/commits/70ab8e2a329bc993570c10809b55645d80a3d384)

File: ethpm_types/base.py [hunk 4]
```diff
@@ -33,3 +37,21 @@ def json(self, *args, **kwargs) -> str:
             kwargs["exclude_none"] = True
 
         return super().json(*args, **kwargs)
+
+    @classmethod
+    @no_type_check
+    def _get_value(
```
Comment (fubuloubu) [inline ethpm_types/base.py:43]: This won't work across the board unless we subclass this class then?

Comment (NotPeopling2day) [reply ethpm_types/base.py:43]: Correct, every model that would need to validate Hexbytes would need to either use the basemodel from this library or add this method to their pydantic basemodel. The other piece is to use our specific version of HexBytes with the validation methods added.

File: tests/test_hexbytes.py (hunk 3 above)
Comment (unparalleled-js) [reply tests/test_hexbytes.py:3]: Ah, this seems good then. I suppose that would not change anything.

**Review by unparalleled-js — APPROVED** [review]
LGTM
Once @fubuloubu 's concern about working across the board is addressed

File: ethpm_types/base.py [hunk 5]
```diff
@@ -33,3 +37,21 @@ def json(self, *args, **kwargs) -> str:
...
+        cls,
+        v: Any,
+        *args,
+        **kwargs,
+    ) -> Any:
+
+        if isinstance(v, HexBytes):
```
Comment (unparalleled-js) [inline ethpm_types/base.py:50]: @NotPeopling2day Is this line tested? This is what I am talking about on Tele

Comment (NotPeopling2day) [reply ethpm_types/base.py:50]: Yes, when the `.dict()` method is called the value of the member will be HexBytes

**Comment by unparalleled-js** [comment]
What I am trying to understand:

What do we do when web3.py or whomever gives Ape a bunch of hexbytes.HexBytes.. Do we have to first convert those to ethpm-types.HexBytes or can ethpm-types handle this conversion somewhere / somehow on its own?

**Pull request merged**